pkg/kube/resources.go

```diff
@@ -54,18 +50,12 @@ func CreateResourceProvider(directory string) (*ResourceProvider, error) {
 // CreateResourceProviderFromPath returns a new ResourceProvider using the YAML files in a directory
 func CreateResourceProviderFromPath(directory string) (*ResourceProvider, error) {
 	resources := ResourceProvider{
-		ServerVersion:          "unknown",
-		SourceType:             "Path",
-		SourceName:             directory,
-		Nodes:                  []corev1.Node{},
-		Deployments:            []appsv1.Deployment{},
-		StatefulSets:           []appsv1.StatefulSet{},
-		DaemonSets:             []appsv1.DaemonSet{},
-		Jobs:                   []batchv1.Job{},
-		CronJobs:               []batchv1beta1.CronJob{},
-		ReplicationControllers: []corev1.ReplicationController{},
-		Namespaces:             []corev1.Namespace{},
-		Pods:                   []corev1.Pod{},
+		ServerVersion: "unknown",
+		SourceType:    "Path",
+		SourceName:    directory,
+		Nodes:         []corev1.Node{},
+		Namespaces:    []corev1.Namespace{},
+		Pods:          []corev1.Pod{},
```

Comment:
Rather than `Pods`, maybe it makes sense for this to be `Controllers: []GenericController{}`. Then we can do all the Owner resolution here, instead of inside `Validate`.

If possible, I'd like to keep interactions w/ the k8s API here, rather than passing around an interface.